src/flask_uploads/flask_uploads.py

```diff
@@ -307,39 +305,29 @@ def save(
         """
         if not isinstance(storage, FileStorage):
             raise TypeError("storage must be a werkzeug.FileStorage")
-
-        if folder is None and name is not None and "/" in name:
-            folder, name = os.path.split(name)
         if storage.filename is None:
             raise ValueError("Filename must not be empty!")
         basename = self.get_basename(storage.filename)
-
         if not self.file_allowed(storage, basename):
             raise UploadNotAllowed()
 
-        if name:
-            if name.endswith('.'):
-                basename = name + extension(basename)
-            else:
-                basename = name
+        parts = []
+        if folder is not None:
+            parts.append(folder)
+        parts.append(basename if name is None else name)
+        target = self.config.destination.joinpath(*parts)
 
-        if folder:
-            target_folder = os.path.join(self.config.destination, folder)
-        else:
-            target_folder = self.config.destination
-        if not os.path.exists(target_folder):
-            os.makedirs(target_folder)
-        if os.path.exists(os.path.join(target_folder, basename)):
-            basename = self.resolve_conflict(target_folder, basename)
-
-        target = os.path.join(target_folder, basename)
-        storage.save(target)
-        if folder:
-            return posixpath.join(folder, basename)
-        else:
-            return basename
+        if not target.suffix and target.stem.endswith('.'):
+            target = target.with_name(f"{target.stem}{extension(basename)}")
+        # TODO: target.parent.mkdir(parents=True, exist_ok=True)
+        os.makedirs(target.parent)
+        if os.path.exists(str(target)):
+            target = self.resolve_conflict(target)
+
+        storage.save(str(target))
```

Comment:
The Werkzeug filestorage type can use a Path object as target, see https://werkzeug.palletsprojects.com/en/1.0.x/datastructures/#werkzeug.datastructures.FileStorage.save